File: python/2270. Number of Ways to Split Array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        n = len(nums)
        pref_sum = [0] * n
        prev = 0
        for i in range(n):
            pref_sum[i] = prev + nums[i]
            prev = pref_sum[i]
        ans = 0
        tot = pref_sum[-1]
        for i in range(n-1):
            left_sum = pref_sum[i]
            right_sum = tot - left_sum
            if left_sum >= right_sum:
                ans += 1
        return ans

File: rust/2270. Number of Ways to Split Array.rs
impl Solution {
    pub fn ways_to_split_array(nums: Vec<i32>) -> i32 {
        let mut running_sum = 0i64;
        let total: i64 = nums.iter().map(|&x| x as i64).sum();
        
        nums.iter()
            .take(nums.len() - 1)
            .filter(|&&x| {
                running_sum += x as i64;
                running_sum >= total - running_sum
            })
            .count() as i32
    }
}
